core/templates/dev/head/pages/library/library.html

```diff
@@ -27,6 +27,7 @@
     GLOBALS.SEARCH_DROPDOWN_CATEGORIES = JSON.parse(
       '{{SEARCH_DROPDOWN_CATEGORIES|js_string}}');
     GLOBALS.PAGE_MODE = JSON.parse('{{page_mode|js_string}}');
+    GLOBALS.hasFullyRegistered = JSON.parse('{{has_fully_registered|js_string}}');
```

Comment:
Looking at the code, I'm not actually sure why hasFullyRegistered is relevant. I suggest we just use the existing Boolean(GLOBALS.username) to determine whether or not the user is already logged in, and not pass has_fully_registered at all to the frontend if it's not needed here.